bigtable-hbase-1.x-parent/bigtable-hbase-1.x-benchmarks/src/main/java/com/google/cloud/bigtable/hbase1_x/BenchmarkDataUtils.java

```diff
@@ -0,0 +1,130 @@
+    return BigtableConfiguration.connect(config);
+  }
+
+  static void createAndFillTable(Connection connection, BenchmarkJobConfig jobConfig)
+      throws IOException {
+
+    try (Admin admin = connection.getAdmin()) {
+      if (admin.tableExists(jobConfig.tableName)) {
+        LOG.info("Using existing table");
+        // Assuming the existing table would have all required data
+        return;
+      }
+      LOG.info("Did not found the the table with tableName as: " + jobConfig.tableName);
+
+      admin.createTable(
+          new HTableDescriptor(jobConfig.tableName)
+              .addFamily(new HColumnDescriptor(COL_FAMILY).setMaxVersions(1)));
+    }
+
+    try (BufferedMutator bufferedMutator = connection.getBufferedMutator(jobConfig.tableName)) {
+
+      String rowPrefix = getRowPrefix(jobConfig.cellsPerRow, jobConfig.cellSize);
+      ImmutableList.Builder<Put> putBuilder = ImmutableList.builderWithExpectedSize(ROWS_PER_BATCH);
+      final long currentTimestamp = System.currentTimeMillis();
+
+      for (int rowInd = 0, sendBatchInd = 0;
+          rowInd < jobConfig.totalRows;
+          rowInd++, sendBatchInd++) {
+
+        // zero padded row-key
+        Put put = new Put(Bytes.toBytes(rowPrefix + String.format(SUFFIX_FMT, rowInd)));
+
+        for (int cellInd = 0; cellInd < jobConfig.cellsPerRow; cellInd++) {
+          put.addColumn(
+              COL_FAMILY,
+              Bytes.toBytes("qualifier-" + String.format(SUFFIX_FMT, cellInd)),
+              currentTimestamp,
+              getRandomBytes(jobConfig.cellSize));
+        }
+        putBuilder.add(put);
+
+        if (sendBatchInd % ROWS_PER_BATCH == 0 || rowInd == (jobConfig.totalRows - 1)) {
+          bufferedMutator.mutate(putBuilder.build());
+
+          // Does this makes sense?
+          bufferedMutator.flush();
+
+          putBuilder = ImmutableList.builderWithExpectedSize(ROWS_PER_BATCH);
+        }
+      }
+    }
+    LOG.info(String.format("Mutate table with %d rows", jobConfig.totalRows));
+  }
+
+  static String getTableId(int cellsPerRow, int cellSize) {
+    return String.format("benchmark_table_%d-cells_%d-bytes", cellsPerRow, cellSize);
+  }
+
+  static String getRowPrefix(int cellsPerRow, int cellSize) {
+    return String.format("key_%d-cell_%d-bytes-", cellsPerRow, cellSize);
+  }
+
+  static byte[] getRandomBytes(int size) {
+    byte[] data = new byte[size];
+    new Random().nextBytes(data);
+    return data;
+  }
+
+  static void createTableToWrite(Connection connection, TableName writeToTableName)
+      throws IOException {
+    try (Admin admin = connection.getAdmin()) {
+
+      if (admin.tableExists(writeToTableName)) {
+        // Deleting the existing table to create a fresh table to write to.
+        admin.deleteTable(writeToTableName);
+      }
+      admin.createTable(
+          new HTableDescriptor(writeToTableName)
+              .addFamily(new HColumnDescriptor(COL_FAMILY).setMaxVersions(1)));
+    }
```

Comment:
@igorbernstein2 Do you think `Admin#truncateTable `of an existing table is better than deleting and creating `benchmark_write` table?

Reply:
I don't think we need to bother with truncation. Just set a GC Policy to keep only the latest version and constrain the key space to < 1 million

Reply:
I tried to add a constraint to it, but it seems we need to include `hbase-server` dependencies to get `BaseConstraint`. I was planning to follow [this(hbase constraints docs)](https://hbase.apache.org/devapidocs/org/apache/hadoop/hbase/constraint/package-summary.html#usage) but thought to confirm with you before including these dependencies.

Reply:
I'm not sure how BaseConstraint is relevant here. I just meant the mutation benchmarks should do something like:
`String.format("some-const-prefix-%d", random.nextInt(1_000_000))`